rust/phoenix-sdk/src/utils.rs

```
@@ -0,0 +1,91 @@
+
+    let mut instructions = vec![];
+
+    // Include claim seat instruction if error finding account or data is empty.
+    if let Ok(seat_account) = seat_account {
+        if seat_account.data.is_empty() {
+            instructions.push(create_claim_seat_instruction(trader, market_pubkey));
+        }
+    } else {
+        instructions.push(create_claim_seat_instruction(trader, market_pubkey));
+    }
+
+    let market_bytes = client.get_account_data(market_pubkey).await?;
+    let (header_bytes, market_bytes) = market_bytes.split_at(size_of::<MarketHeader>());
+    let market_header = bytemuck::try_from_bytes::<MarketHeader>(header_bytes)
+        .map_err(|e| anyhow::anyhow!("Error deserializing market header. Error: {:?}", e))?;
+
+    let market =
+        dispatch_market::load_with_dispatch(&market_header.market_size_params, market_bytes)?.inner;
+
+    let trader_tree = market.get_registered_traders();
+    let num_traders = trader_tree.len();
+    let max_traders = trader_tree.capacity();
+
+    if num_traders == max_traders {
+        //Find a seat to evict (a trader with no locked base or quote lots) and evict trader.
+        for (trader_pubkey, trader_state) in trader_tree.iter() {
+            if trader_state.base_lots_locked == 0 && trader_state.quote_lots_locked == 0 {
+                let evict_trader_state = EvictTraderAccountBackup {
+                    trader_pubkey: *trader_pubkey,
+                    base_token_account_backup: None,
+                    quote_token_account_backup: None,
+                };
+                let evict_trader_ix = create_evict_seat_instruction(
+                    market_pubkey,
+                    &market_header.base_params.mint_key,
+                    &market_header.quote_params.mint_key,
+                    trader,
+                    vec![evict_trader_state],
+                );
+                instructions.push(evict_trader_ix);
+                break;
+            }
```

if not found, error

---

Makes sense. To be clear, the case we want to return an error is if the trader state is full but we cannot find a trader with no locked lots to evict (inside the for loop).